projects/fsastorefrontlib/src/core/my-account/facade/quote.service.ts

```diff
@@ -137,31 +135,28 @@ export class QuoteService {
   }
 
   protected loadChooseCoverForm(insuranceQuote: any, categoryCode: string) {
-    if (
-      insuranceQuote &&
-      insuranceQuote.quoteDetails &&
-      insuranceQuote.quoteDetails.entry
-    ) {
-      const dataId = insuranceQuote.quoteDetails.entry
-        .filter(details => details.key === 'formId')
-        .map(mapEntry => mapEntry.value)[0];
+    if (insuranceQuote && insuranceQuote.quoteDetails) {
+      const dataId = insuranceQuote.quoteDetails.formId;
       this.formDataService.loadFormData(dataId);
-      this.formDataService
-        .getFormData()
-        .pipe(
-          map(formData => {
-            if (formData.formDefinition) {
-              this.formDataStorageService.setFormDataToLocalStorage({
-                id: dataId,
-                formDefinition: {
-                  formId: formData.formDefinition.formId,
-                },
-                categoryCode: categoryCode,
-              });
-            }
-          })
-        )
-        .subscribe();
+      this.subscription.add(
```

Comment:
I know this approach was already implemented, but since we are refactoring this, maybe we should refactor this approach as well.